filters/auth/oidc.go

```diff
@@ -698,7 +701,20 @@ func (f *tokenOidcFilter) Request(ctx filters.FilterContext) {
 			return
 		}
 		// 1. Client prepares an Authentication Request containing the desired request parameters.
-		f.doOauthRedirect(ctx)
+		// clear existing, invalid cookies
+		var purgeCookies = make([]*http.Cookie, len(cookies))
+		for i, c := range cookies {
+			purgeCookies[i] = &http.Cookie{
+				Name:     c.Name,
+				Value:    "",
+				Path:     "/",
+				Domain:   extractDomainFromHost(ctx.Request().Host, 1),
+				MaxAge:   -1,
+				Secure:   true,
+				HttpOnly: true,
+			}
```

Comment:
Why not copy everything from `c` and set empty `Value` and `MaxAge`?
Or maybe even modify `cookies` values in-place and it to `f.doOauthRedirect`?..

Reply:
The [request cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cookie) contains only `Name` and `Value`. In order to clear we need empty Value, Path, Domain, MaxAge and maybe Secure and HttpOnly. Sure we can reuse the cookie instance but only the Name is relevant the rest we have to set